sky/jobs/controller.py

```diff
@@ -334,6 +334,13 @@ async def _run_one_task(self, task_id: int, task: 'sky.Task') -> bool:
             resources_str = backend_utils.get_task_resources_str(
                 task, is_managed_job=True)
 
+            # Get full_resources_json using to_yaml_config from task resources
+            full_resources_json = None
+            if task.resources:
+                # Get the first Resources object from the set/list
+                task_resource = next(iter(task.resources))
```

Comment:
Theoretically, we should get whichever resource fits on the cluster we chose. But at this point we haven't chosen the cluster yet, so this is obviously impossible.
I think it's fine to just say "this doesn't work with multiple resources (multiple GPU options/ordered/any_of) right now", but it is a limitation.
We could also maybe just serialize all the resources for the task - see `_resources_to_config` in task.py. Or set the resources once we have actually launched? (but this seems race-prone)